sys/include/net/gnrc/rpl/structs.h

```diff
@@ -238,7 +238,24 @@ struct gnrc_rpl_parent {
 typedef struct {
     uint16_t ocp;   /**< objective code point */
     uint16_t (*calc_rank)(gnrc_rpl_parent_t *parent, uint16_t base_rank); /**< calculate the rank */
-    gnrc_rpl_parent_t *(*which_parent)(gnrc_rpl_parent_t *, gnrc_rpl_parent_t *); /**< compare for parents */
+    gnrc_rpl_parent_t *(*which_parent)(gnrc_rpl_parent_t *, gnrc_rpl_parent_t *); /**< retrieve the better parent */
+
+    /**
+     * @brief   Compare two @ref gnrc_rpl_parent_t.
+     *
+     * This function is used to determine the parent list order. The parents
+     * are ordered from the preferred parent to the least preferred parent.
+     * This function should compare two parents based on the criteria of the
+     * objective function.
+     *
+     * @param[in] parent1 First parent to compare.
+     * @param[in] parent2 Second parent to compare.
+     *
+     * @return      Zero if the parents are of equal preference,
+     *              positive result if the second parent is prefered,
+     *              negative result if the first parent is prefered.
+     */
+    int (*parent_cmp)(gnrc_rpl_parent_t *parent1, gnrc_rpl_parent_t *parent2);
```

Comment:
IMHO parameter names `p1` and `p2` were fine, too - and match with actual implementation in `of0.c` below.